cache_test.go

```diff
@@ -173,3 +173,28 @@ func TestRoundRobinShuffle(t *testing.T) {
 	require.Equal(t, net.IP{0, 0, 0, 2}, a1.A)
 	require.Equal(t, net.IP{0, 0, 0, 1}, a2.A)
 }
+
+// Truncated responses should not be cached
+func TestCacheNoTruncated(t *testing.T) {
+	var ci ClientInfo
+	q := new(dns.Msg)
+	r := &TestResolver{
+		ResolveFunc: func(q *dns.Msg, ci ClientInfo) (*dns.Msg, error) {
+			a := new(dns.Msg)
+			a.SetReply(q)
+			a.Truncated = true
+			return a, nil
+		},
+	}
+
+	c := NewCache("test-cache", r, CacheOptions{})
+
+	// Both queries should hit the upstream resolver
+	q.SetQuestion("test.com.", dns.TypeA)
```

Comment:
This domain `test.com` worries me unless you own the domain.
I would either query `www.google.com` or `github.com` for long term change and whitelist reasons. Those domains are much more "stable" as of today than `test.com` which I believe will exchange hands and possibly go `NXDOMAIN`, or `SERVFAIL` if push comes to shove. To add to that in my config I use pihole as the root lightweight blocker. If someone has `test.com` in there  base level DNS then that could add an issue as well depeneding on there config and setup if its blocked or not.

Reply:
Good point. While these queries don't actually hit any public resolvers, it's not good practice in general. I switched to using example.com instead which is intended to be used for testing.

Reply:
I just checked test.com is blocked on my config